### PR TITLE
Getting_Involved/Newcomers.rst: Add a better newcomer page

### PR DESCRIPTION
**Pull request by Adrianzatreanu**
This page will ultimately be sent to all of the newcomers and
it will guide them appropriately, so they can start contributing
just by following the guide on this page.

Fixes https://github.com/coala-analyzer/coala/issues/1734

### CODE REVIEW

File: docs/Getting_Involved/Newcomers.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+Start working
+-------------
+
+Now is the time to pick an issue.
+Here are two links that will lead you to newcomers issues:
+
+- For coala project: `coala newcomer issues <https://github.com/coala-analyzer/coala/issues?q=is%3Aissue+is%3Aopen+label%3Adifficulty%2Fnewcomer>`_
+
+- Also, for our coala-bears: `coala-bears newcomer issues <https://github.com/coala-analyzer/coala-bears/issues?q=is%3Aissue+is%3Aopen+label%3Adifficulty%2Fnewcomer>`_
```
Comment (sils1297) [inline docs/Getting_Involved/Newcomers.rst]: can you use the generic link I gave you as that inclues newcomer issues for *all* projects?

Comment (sils1297) [inline docs/Getting_Involved/Newcomers.rst]: https://github.com/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+user%3Acoala-analyzer+label%3Adifficulty%2Fnewcomer

File: docs/Getting_Involved/Newcomers.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+- your Pull Request gets accepted, and your commit will get merged into the
+master branch;
+- your Pull Request doesn't, and therefore you will need to edit your code
+or your commit message;
+
+.. note::
+    See also: `Review Process <http://coala.readthedocs.org/en/latest/Getting_Involved/Review.html>`_
+
+Most of the time it is the second option. Now if you need your code, you can
+simply edit it again, add it using
```
Comment (sils1297) [inline docs/Getting_Involved/Newcomers.rst]: yeah we should stress this a bit more! "It's highly unlikely that your PR will be accepted on the first attempt - but don't worry that's just how it works. It helps us keeping coala clean and stable." something like that?

File: docs/Getting_Involved/Newcomers.rst [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+::
+
+    $ git add <file_name>
+
+and then using
+::
+
+    $ git commit --amend
```
Comment (sils1297) [inline docs/Getting_Involved/Newcomers.rst]: just let them do `git commit -a --amend`

File: docs/Getting_Involved/Newcomers.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+fine by our reviewers, you can simply send it again like this. If not, edit it
+and send it. You have successfully edited your last commit!
+
+.. note::
+    Don't forget! After editing your commit, you will have to push it again.
+    This can be done using:
+
+::
+
+    $ git push --force <your_fork_name>
```
Comment (sils1297) [inline docs/Getting_Involved/Newcomers.rst]: we should have a section with contents like:

- congrats, your PR got accepted
- you're awesome
- try writing a simple bear possibly, those are rewarding

File: docs/Getting_Involved/Newcomers.rst [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+or your commit message;
+
+It's highly unlikely that your PR will be accepted on the first attempt - but
+don't worry that's just how it works. It helps us keeping coala clean and stable.
+
+.. note::
+    See also: `Review Process <http://coala.readthedocs.org/en/latest/Getting_Involved/Review.html>`_
+
+Most of the time it is the second option. Now if you need your code, you can
+simply edit it again, add it and commit it using
```
Comment (sils1297) [inline docs/Getting_Involved/Newcomers.rst]: doesn't fit anymore

Comment (Adrianzatreanu) [inline docs/Getting_Involved/Newcomers.rst]: why not? if that command does both of them, shouldnt this be good?

Comment (sils1297) [inline docs/Getting_Involved/Newcomers.rst]: "Most of the time it is the second option" was already mentioned before this reads weird

Comment (Adrianzatreanu) [inline docs/Getting_Involved/Newcomers.rst]: ok i get what u mean now.

**Comment by Adrianzatreanu** [comment]
Ok, i will fix the rst issues and open newcomer issues for grammar mistakes :) thanks

Also, for me, seealso didnt work, i might have used it wrong or something, i will check it later today

**Comment by AbdealiJK** [comment]
seealso is used in other places:
```
ajk@ajk-ThinkPad-R60:~/Documents/coala/coala$ git grep "seealso"
docs/Getting_Involved/Writing_Good_Commits.rst:.. seealso::
docs/Getting_Involved/Writing_Tests.rst:    .. seealso::
docs/Users/Git_Hooks.rst:.. seealso::
```
Check them out ?

File: docs/Getting_Involved/Newcomers.rst [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+Things to do before pushing
+---------------------------
+
+Now you need to make sure your change is actually working. For this, you will
+need to test it locally before pushing it to your fork, and checking it with
+concrete examples. So basically, run tests.
+
+.. seealso::
+
+    `Executing tests <http://coala.readthedocs.org/en/latest/Getting_Involved/Testing.html>`_
```
Comment (sils1297) [inline docs/Getting_Involved/Newcomers.rst]: and coala should be run too actually, right?

File: docs/Getting_Involved/Newcomers.rst [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+**Congratulations!** You have now created your first Pull Request!
+
+What to do after creating a PR
+------------------------------
+
+After creating your Pull Request, there's two options:
+
+- your Pull Request gets accepted, and your commit will get merged into the
+  master branch
+
```
Comment (sils1297) [inline docs/Getting_Involved/Newcomers.rst]: why this newline?

Comment (Adrianzatreanu) [inline docs/Getting_Involved/Newcomers.rst]: newline between list items are optional, they do nothing. we can add a newcomer issue for this if you want :)

Comment (Adrianzatreanu) [inline docs/Getting_Involved/Newcomers.rst]: more redundant things like these = more newcomer issues

**Comment by sils1297** [comment]
ack b8e4d059e348b31ef22651241b585f2667206904

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by AbdealiJK** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by sils1297** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
@rultor merge

**Comment by Adrianzatreanu** [comment]
reack 9b788a2

**Comment by AbdealiJK** [comment]
@rultor merge

**Comment by AbdealiJK** [comment]
Merged as I wanted to use it :)

**Comment by Adrianzatreanu** [comment]
awesome :)

**Pull request merged**